### PR TITLE
Refs #36981 - Add structure upload for deb repositories

### PR DESCRIPTION
**Pull request by Manisha15**

### CODE REVIEW

**Comment by sjha4** [comment]
@Manisha15 : Could you add some testing steps here and a little background around the change. Remember similar changes in pulp which have landed in 4.11. Not sure if you have a release preference for this as well.

**Comment by quba42** [comment]
@sjha4 For now, these changes are mostly invisible to Katello users. The thing to test is to create a (empty) deb type repository and upload any random .deb package to it using the GUI. For example, you could use `wget ftp.de.debian.org/debian/pool/main/v/vim/vim_9.0.1378-2_amd64.deb`.

Once the package has uploaded successfully you can open the "Published At" link from the repository page to see the publication being served by Pulp. If you navigate to the `dists/` folder in the published repo, you will see both a `katello/` and a `default/` sub-folder. Without the changes from this PR, you will see only the `default/` sub-folder in this location. The extra folder comes from the structure content that pulp_deb has created because of the change in upload API endpoint that this PR introduces.

**Comment by quba42** [comment]
Thought of one more possible thing to test: Uploading the same .deb package into two different Katello repositories. I have a vague memory there once was a bug that caused the second upload to fail in such cases. Pretty sure we did already test this for this PR though.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html [hunk 1]
```diff
@@ -52,14 +52,6 @@ <h4 translate>Publishing Settings</h4>
...
               on-save="save(repository)">
           </dd>
-
-          <dt ng-if="product.redhat != true" translate>Metadata Expiration (Seconds)</dt>
-          <dd bst-edit-number="repository.metadata_expire" on-save="save(repository)" readonly="product.redhat || denied('edit_products', product)"
-              deletable="repository.metadata_expire !== null"
-              on-delete="clearMetadataExpire()"
-              min="1"
-          >
-          </dd>
```
Comment (sjha4) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html:62]: Why are we removing this?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html [hunk 2]
```diff
@@ -263,15 +262,6 @@ <h4 translate>Sync Settings</h4>
         </dd>
       </span>
 
-      <span ng-if="repository.content_type === 'yum'">
-        <dt translate>Ignore treeinfo</dt>
```
Comment (sjha4) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html:267]: Why are we removing this?

Comment (Manisha15) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-info.html]: Fixed it, it was not needed to remove.

**Review by sjha4 — APPROVED** [review]
Code looks good. I was able to test the UI changes and the upload flow.
Empty_url_message
package-upload

Let me know when you want this merged and I can merge this. Thanks @Manisha15 , @quba42 🎉

**Pull request merged**